### PR TITLE
Add 3 theme modes

### PR DESCRIPTION
**Pull request by azizramdan**
Hi @nyancodeid, I have added 3 theme modes; system theme mode, light mode, and dark mode.

This PR can solve #86

### CODE REVIEW

**Comment by nyancodeid** [comment]
Awesome, but toggle theme in 3 mode, how this possible with only one button? as i understand toggle is about 2 value switch on / off, true or false, dark or light. I think we keep dark mode toggle only for 2 mode, light and dark at this moment.

**Comment by azizramdan** [comment]
@nyancodeid check laravel docs for example, https://laravel.com/docs/8.x

**Comment by nyancodeid** [comment]
Interesting, but it's very confusing at first because it's not like how the toggle button actually works.

File: src/components/AppHero.vue [hunk 1]
```diff
@@ -1,29 +1,89 @@
...
     <div class="absolute right-4 top-4">
-      <button class="p-2 border rounded-full text-gray-900 dark:text-gray-100" @click="toggleDarkMode">
-        <i-ri-sun-line v-show="isDark" />
-        <i-ri-moon-line v-show="!isDark" />
+      <button 
+        class="p-2 border rounded-full text-gray-900 dark:text-gray-100" 
+        :title="titleTheme"
+        @click="toggleTheme"
+      >
+        <i-bi-circle-half v-if="theme === 'system'" />
```
Comment (nyancodeid) [inline src/components/AppHero.vue:84]: for the system icon can you change using Material Design Icon `<i-mdi-circle-half-full />`, and the moon stars we prefer from Remix Icons `<i-ri-moon-clear-line />`

**Comment by nyancodeid** [comment]
For system preferred mode did we have some event listener if we changed our dark mode from our operating system settings? @azizramdan

File: src/components/AppHero.vue (hunk 1 above)
Comment (azizramdan) [reply src/components/AppHero.vue]: done

Comment (nyancodeid) [reply src/components/AppHero.vue]: Thank you

**Comment by nyancodeid** [comment]
> For system preferred mode did we have some event listener if we changed our dark mode from our operating system settings? @azizramdan

Nevermind, we can think about it later on. Thanks @azizramdan i'm gonna merge this PR. Awesome, good job

**Comment by azizramdan** [comment]
> > For system preferred mode did we have some event listener if we changed our dark mode from our operating system settings? @azizramdan
> 
> Nevermind, we can think about it later on. Thanks @azizramdan i'm gonna merge this PR. Awesome, good job

We don't have any events yet, we need to manually refresh page to get system theme

https://user-images.githubusercontent.com/27954794/136049420-a53f7a8d-ce75-4c9b-9ca5-21d642e5df53.mp4

**Comment by nyancodeid** [comment]
Got it, let me know if you can make that feature.

**Pull request merged**